langchain/src/tools/ical.ts

```diff
@@ -0,0 +1,115 @@
+import { z } from "zod";
+
+import { BaseLLM } from "../llms/base.js";
+import { PromptTemplate } from "../prompts/prompt.js";
+import { Tool, ToolParams } from "./base.js";
+import { StructuredOutputParser } from "../output_parsers/structured.js";
+import { BaseFileStore } from "../schema/index.js";
+
+interface ICalParams extends ToolParams {
+  store: BaseFileStore;
+  llm: BaseLLM;
+}
+
+export class ICalTool extends Tool {
+  name = "iCal";
+
+  description =
+    "Useful for creating iCal calendar events. Input should be an instruction describing what event(s) to create.";
+
+  eventsSchema = z.object({
+    name: z.string().describe("Name of the event event"),
+    events: z.array(
+      z.object({
+        start: z.string().describe("Start date of the event"),
+        end: z.string().describe("Env date of the event"),
+        summary: z.string().describe("Summary of the event"),
+        description: z.string().describe("Description of the event"),
+        location: z
+          .string()
+          .describe("Location of the event if any")
+          .optional(),
+        url: z.string().describe("URL of the event if any").optional(),
+      })
+    ),
+  });
+
+  store: BaseFileStore;
+
+  llm: BaseLLM;
+
+  constructor({ store, llm, verbose, callbacks }: ICalParams) {
+    super(verbose, callbacks);
+    this.store = store;
+    this.llm = llm;
+  }
+
+  /** @ignore */
+  async _call(input: string): Promise<string> {
+    try {
+      const { name, events } = await this.getEventsObject(input);
+      const { ical } = await this.imports();
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
AFAICT ical-generator does provide typescript definitions, no need to add @ts-ignore

Reply:
I had to add that because of `ts-jest`. It strangely doesn't seem to pick up the types. 

<img width="1047" alt="Screenshot 2023-05-12 at 00 37 54" src="https://github.com/hwchase17/langchainjs/assets/6696862/379f431f-681c-49c8-8f2c-81c4462c67ac">

Reply:
Understandable, let's keep it that way.